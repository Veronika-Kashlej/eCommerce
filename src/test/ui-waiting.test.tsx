import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import WaitingModal from '@/components/waiting/Waiting';

describe('WaitingModal', () => {
  it('should not render when isOpen is false', () => {
    const { container } = render(<WaitingModal isOpen={false} />);
    expect(container.firstChild).toBeNull();
  });

  it('should render modal when isOpen is true', () => {
    render(<WaitingModal isOpen={true} />);

    const backdrop = screen.getByTestId('waiting-backdrop');
    const modal = screen.getByTestId('waiting-modal');
    const icon = screen.getByText('⏳');

    expect(backdrop).toBeInTheDocument();
    expect(modal).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(modal).toHaveAttribute('aria-live', 'polite');
    expect(modal).toHaveAttribute('aria-busy', 'true');
    expect(modal).toHaveAttribute('role', 'status');
  });

  it('should match snapshot when open', () => {
    const { asFragment } = render(<WaitingModal isOpen={true} />);
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="waiting-backdrop"
          data-testid="waiting-backdrop"
        >
          <div
            aria-busy="true"
            aria-live="polite"
            class="waiting-modal"
            data-testid="waiting-modal"
            role="status"
          >
            ⏳
          </div>
        </div>
      </DocumentFragment>
    `);
  });

  it('should match snapshot when closed', () => {
    const { asFragment } = render(<WaitingModal isOpen={false} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
