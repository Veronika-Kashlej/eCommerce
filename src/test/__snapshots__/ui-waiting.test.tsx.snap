// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WaitingModal > should match snapshot when closed 1`] = `<DocumentFragment />`;

exports[`WaitingModal > should match snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="waiting-backdrop"
  >
    <div
      class="waiting-modal"
    >
      ‚è≥
    </div>
  </div>
</DocumentFragment>
`;
